src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -61,7 +61,9 @@ allocation_constraints default_constraints() {
 }
 
 result<std::vector<model::broker_shard>>
-partition_allocator::allocate_partition(partition_constraints p_constraints) {
+partition_allocator::allocate_partition(
```

Comment:
can we use `reallocate_partition` api to do this, just adding the code for rack awearnes should be enough

Reply:
It that case we should skip adding distinct_rack constraint in allocate_partition.
Because if we add multiple soft constraints, we decrease weight of one.
I can't come up with idea, how to do it. Because we cannot check that constraints contain rack awarreness.

Reply:
i do not understand the question, we should be able to add any number of constraints into reallocate partitions, why is this impossible ?

Reply:
now i see it, thank you for explaining